# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../libdvb/libdvb_x86_64.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- libdvb-0.5.5.1/sample_progs/cam_menu.cc.orig	2005-09-08 10:55:29.000000000 -0700
+++ libdvb-0.5.5.1/sample_progs/cam_menu.cc	2005-09-08 11:08:25.000000000 -0700
@@ -46,7 +46,13 @@
 
 void byebye(int i, void *f)
 {
-	int fd = (int) f;
+	int fd;
+	if (sizeof(void *) == sizeof(int))
+		fd = (long) f;
+	else {
+		long fd_long = (long) f;
+		fd = (int) fd_long;
+	}
 
 	CamPacket cp(fd, CMENU_CLOSE, 0, NULL);
 	cp.psend();
